WalletWasabi.Fluent/Controls/TagsBox.xaml

```diff
@@ -0,0 +1,102 @@
+        xmlns:c="clr-namespace:WalletWasabi.Fluent.Controls">
+    <Style Selector="c|TagsBox">
+        <Setter Property="Foreground" Value="{DynamicResource TextControlForeground}" />
+        <Setter Property="Background" Value="{DynamicResource TextControlBackground}" />
+        <Setter Property="BorderBrush" Value="{DynamicResource TextControlBorderBrush}" />
+        <Setter Property="BorderThickness" Value="{DynamicResource TextControlBorderThemeThickness}" />
+        <Setter Property="FontSize" Value="{DynamicResource ControlContentThemeFontSize}" />
+        <Setter Property="MinHeight" Value="{DynamicResource TextControlThemeMinHeight}" />
+        <Setter Property="MinWidth" Value="{DynamicResource TextControlThemeMinWidth}" />
+        <Setter Property="Padding" Value="{DynamicResource TextControlThemePadding}" />
+        <Setter Property="Focusable" Value="True" />
+        <Setter Property="Cursor" Value="IBeam" />
+        <Setter Property="Template">
+            <ControlTemplate>
+                <DataValidationErrors>
+                    <Border x:Name="PART_Border" Background="{TemplateBinding Background}"
+                            BorderBrush="{TemplateBinding BorderBrush}"
+                            BorderThickness="{TemplateBinding BorderThickness}"
+                            Padding="{TemplateBinding Padding}" Focusable="False">
+                        <ItemsPresenter Name="PART_ItemsPresenter"
+                                        Items="{TemplateBinding Items}"
+                                        ItemTemplate="{TemplateBinding ItemTemplate}">
+
+                            <ItemsPresenter.ItemsPanel>
+                                <ItemsPanelTemplate>
+                                    <c:ConcatenatingWrapPanel>
+                                        <c:ConcatenatingWrapPanel.ConcatenatedChildren>
+                                            <AutoCompleteBox x:Name="PART_AutoCompleteBox"
+                                                             MinimumPrefixLength="1"
+                                                             IsTextCompletionEnabled="True"
+                                                             Items="{Binding #PART_Border.TemplatedParent.Suggestions}" />
+                                        </c:ConcatenatingWrapPanel.ConcatenatedChildren>
+                                    </c:ConcatenatingWrapPanel>
+                                </ItemsPanelTemplate>
+                            </ItemsPresenter.ItemsPanel>
+                        </ItemsPresenter>
+                    </Border>
+                </DataValidationErrors>
+            </ControlTemplate>
+        </Setter>
+    </Style>
+
+    <Style Selector="c|TagsBox">
+        <Setter Property="(ScrollViewer.HorizontalScrollBarVisibility)" Value="Disabled" />
+        <Setter Property="MinHeight" Value="60" />
+        <Setter Property="ItemTemplate">
+            <DataTemplate DataType="x:String">
```

Comment:
You dont need DataType here....

Just remove the textblock and use a ContentPresenter with Content set to {Binding .}.

That way your item template works with any type.